### PR TITLE
tests: update from ethereum/tests

### PR DESCRIPTION
**Pull request by holiman**
This PR updates the tests

### CODE REVIEW

**Comment by holiman** [comment]
So this one caused all travis builds to go out of memory, it seems it happened on the blockchain tests. @winsvega do you know if any tests recently (in the last 6 months or so) are particularly heavy on memory?

**Comment by winsvega** [comment]
Test reorganization happend. Blockchain tests are now has different subfolders. ValidBlocks, InvalidBlocks

https://github.com/ethereum/tests/tree/develop/BlockchainTests

Might be the issue. I've checked with retesteth all state tests pass. Some blockchain tests has consensus issue. And yes, there are tests that make geth instance to keep 1GB of data (presumably json parser issue). But it also might be geth retesteth implementation.

**Comment by holiman** [comment]
> But it also might be geth retesteth implementation.

So our travis CI executor of state tests and blockchain tests does not use retesteth functionality, so that shouldn't be it. I'll look into it

**Comment by holiman** [comment]
I've sorted most issues, and will push a fix that makes it work on CI, where we skip slow-running tests. 
However, some of those skipped actually fail if the `--short` flag is not used. @winsvega , it seems that these are formatted in some quirky way:

```
    --- FAIL: TestBlockchain/ValidBlocks/bcExploitTest/SuicideIssue.json (0.16s)
        init_test.go:231: json: cannot unmarshal string into Go value of type map[common.UnprefixedAddress]core.GenesisAccount in file testdata/BlockchainTests/ValidBlocks/bcExploitTest/SuicideIssue.json
```
```
    --- FAIL: TestBlockchain/ValidBlocks/bcExploitTest/ShanghaiLove.json (0.07s)
        init_test.go:231: json: cannot unmarshal string into Go value of type map[common.UnprefixedAddress]core.GenesisAccount in file testdata/BlockchainTests/ValidBlocks/bcExploitTest/ShanghaiLove.json
```
So both these have something odd about the `pre` section.

**Comment by holiman** [comment]
Ah... so, you just replaced the `postState` with a hash? Well, of course that will fail...

**Comment by winsvega** [comment]
This files are just too huge 15/30 mb. Parser probably going crazy. I run this tests with retesteth + geth just fine. 

So you say this are the only one that fail?
I have some issues with chain reorg tests on geth

**Comment by holiman** [comment]
... but if you suddenly changes one json-struct into a `string`, we need to update the internal parser to detect this. Of course it won't just work... 
I disabled those two now, let's see if it works

**Comment by holiman** [comment]
> So you say this are the only one that fail?
> I have some issues with chain reorg tests on geth

Yes, that's what I'm saying. However, 
```
	// Skip random failures due to selfish mining test.
	bt.skipLoad(`^bcForgedTest/bcForkUncle\.json`)
	bt.skipLoad(`^bcMultiChainTest/(ChainAtoChainB_blockorder|CallContractFromNotBestBlock)`)
	bt.skipLoad(`^bcTotalDifficultyTest/(lotsOfLeafs|lotsOfBranches|sideChainWithMoreTransactions)`)
```

**Comment by winsvega** [comment]
Yes, those one. I don't think some of it should fail due to ambiguous total difficulty. 

Also forgot to mention that big post state is replaced by hash yes.

File: tests/block_test.go [hunk 1]
```diff
@@ -25,24 +25,25 @@ func TestBlockchain(t *testing.T) {
...
 
-	// Still failing tests that we need to look into
-	//bt.fails(`^bcStateTests/suicideThenCheckBalance.json/suicideThenCheckBalance_Constantinople`, "TODO: investigate")
-	//bt.fails(`^bcStateTests/suicideStorageCheckVCreate2.json/suicideStorageCheckVCreate2_Constantinople`, "TODO: investigate")
-	//bt.fails(`^bcStateTests/suicideStorageCheckVCreate.json/suicideStorageCheckVCreate_Constantinople`, "TODO: investigate")
-	//bt.fails(`^bcStateTests/suicideStorageCheck.json/suicideStorageCheck_Constantinople`, "TODO: investigate")
+	// Slow tests
+	bt.slow(`.*bcExploitTest/DelegateCallSpam.json`)
+	bt.slow(`.*bcExploitTest/ShanghaiLove.json`)
+	bt.slow(`.*bcExploitTest/SuicideIssue.json`)
```
Comment (FrankSzendzielarz) [inline tests/block_test.go]: Did I read somewhere a directive to replace 'suicide' references with 'selfdestruct'?

**Pull request merged**